core/imds/src/test/java/software/amazon/awssdk/imds/Ec2MetadataWithApacheClientTest.java

```diff
@@ -0,0 +1,389 @@
+ */
+
+package software.amazon.awssdk.imds;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.put;
+import static com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.http.Fault;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.io.IOException;
+import java.net.URI;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.imds.internal.DefaultEc2Metadata;
+
+/**
+ * Unit Tests to test the Ec2Metadata Client functionality
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class Ec2MetadataWithApacheClientTest {
+
+    private static final String TOKEN_RESOURCE_PATH = "/latest/api/token";
+
+    private static final String TOKEN_HEADER = "x-aws-ec2-metadata-token";
+
+    private static final String EC2_METADATA_TOKEN_TTL_HEADER = "x-aws-ec2-metadata-token-ttl-seconds";
+
+    private static final String EC2_METADATA_ROOT = "/latest/meta-data";
+
+    private static final String AMI_ID_RESOURCE = EC2_METADATA_ROOT + "/ami-id";
+
+    private SdkHttpClient httpClient;
+
+    @Mock
+    private Ec2Metadata ec2Metadata;
+
+    @Rule
+    public WireMockRule mockMetadataEndpoint = new WireMockRule();
+
+    @Before
+    public void methodSetup() {
+        System.setProperty(SdkSystemSetting.AWS_EC2_METADATA_SERVICE_ENDPOINT.property(), "http://localhost:" + mockMetadataEndpoint.port());
+        httpClient = ApacheHttpClient.create();
+    }
+
+    @Test
+    public void get_whenDummyResponseIsReturned(){
+
+        MetadataResponse metadataResponse = new MetadataResponse("IMDS");
+        when(ec2Metadata.get("/ami-id")).thenReturn(metadataResponse);
+        assertThat(ec2Metadata.get("/ami-id").asString()).isEqualTo("IMDS");
+
+    }
+
+    @Test
+    public void verifyEc2Metadata_equalsAndHashcode(){
```

Comment:
This test should go in https://github.com/srsaikumarreddy/aws-sdk-java-v2/blob/saikred/feature/integTests/core/imds/src/test/java/software/amazon/awssdk/imds/Ec2MetadataTest.java

Reply:
Noted.